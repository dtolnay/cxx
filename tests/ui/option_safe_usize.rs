const _: () = cxx::private::assert_option_safe::<usize>();

fn main() {}
