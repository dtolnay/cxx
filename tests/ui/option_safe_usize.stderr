error[E0080]: evaluation of `cxx::private::assert_option_safe::__SizeCheck::<usize>::_IS_NICHE` failed
 --> src/rust_option.rs
  |
  |             assert!(core::mem::size_of::<Option<U>>() == core::mem::size_of::<U>());
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: core::mem::size_of::<Option<U>>() == core::mem::size_of::<U>()', $DIR/src/rust_option.rs:92:13
  |
  = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> src/rust_option.rs
   |
   |     let _: () = __SizeCheck::<T>::_IS_NICHE;
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
